crates/oxc_transformer/src/typescript/enum.rs

```diff
@@ -199,44 +205,34 @@ impl<'a> TypeScriptEnum<'a> {
                 let constant_value =
                     self.computed_constant_value(initializer, &previous_enum_members);
 
+                previous_enum_members.insert(member_name.clone(), constant_value.clone());
+
                 // prev_constant_value = constant_value
                 let init = match constant_value {
                     None => {
                         prev_constant_value = None;
                         let mut new_initializer = ast.move_expression(initializer);
 
-                        // If the initializer is a binding identifier,
-                        // and it is not a binding in the current scope and parent scopes,
-                        // we need to rename it to the enum name. e.g. `d = c` to `d = A.c`
-                        // same behavior in https://github.com/babel/babel/blob/610897a9a96c5e344e77ca9665df7613d2f88358/packages/babel-plugin-transform-typescript/src/enum.ts#L145-L150
-                        let has_binding = matches!(
-                            &new_initializer,
-                            Expression::Identifier(ident) if ctx.scopes().has_binding(ctx.current_scope_id(), &ident.name)
-                        );
-                        if !has_binding {
-                            IdentifierReferenceRename::new(
-                                param_binding.name.clone(),
-                                previous_enum_members.clone(),
-                                ctx,
-                            )
-                            .visit_expression(&mut new_initializer);
-                        }
+                        IdentifierReferenceRename::new(
+                            param_binding.name.clone(),
```

Comment:
Question: any reason not making `Atom` implements `Copy`? `.clone()` always makes me feel the operation is somewhat expensive.

Reply:
I agree. https://github.com/oxc-project/backlog/issues/155

Originally we were considering some form of interning and reference-counting, so we didn't make it `Copy` to leave the door open for that. But now all strings are stored in the arena anyway, so even if we did decide to intern strings, reference-counting would be irrelevant.

Reply:
A PR would be very welcome. But it'd be boring as hell to do! *So many* `.clone()`s to remove!

Reply:
Thanks for both the answer and the backlog link! There are lots of interesting readings over there:)

Reply:
Yeah, it's kind of our "brain dump". We try to keep main issue tracker just for stuff which is being worked on currently, to keep things manageable. But if there's anything in backlog that interests you, please ping me on our [Discord](https://discord.gg/9uXCAwqQZW) and I'll transfer the issue over to main repo so we can discuss.